icu4c/source/i18n/messageformat2_data_model.cpp

```diff
@@ -400,86 +323,46 @@ Option::~Option() {}
 
 Operator::Builder::Builder(UErrorCode& status) : options(OptionMap::Builder(status)) {}
 
-Operator::Builder& Operator::Builder::setReserved(Reserved&& reserved) {
-    isReservedSequence = true;
-    hasFunctionName = false;
-    hasOptions = false;
-    asReserved = std::move(reserved);
-    return *this;
-}
-
 Operator::Builder& Operator::Builder::setFunctionName(FunctionName&& func) {
-    isReservedSequence = false;
-    hasFunctionName = true;
     functionName = std::move(func);
     return *this;
 }
 
 const FunctionName& Operator::getFunctionName() const {
-    U_ASSERT(!isReserved());
-    return std::get_if<Callable>(&contents)->getName();
+    return name;
 }
 
 Operator::Builder& Operator::Builder::addOption(const UnicodeString &key, Operand&& value, UErrorCode& errorCode) noexcept {
     THIS_ON_ERROR(errorCode);
 
-    isReservedSequence = false;
-    hasOptions = true;
     options.add(Option(key, std::move(value)), errorCode);
     return *this;
 }
 
 Operator Operator::Builder::build(UErrorCode& errorCode) {
-    Operator result;
-    if (U_FAILURE(errorCode)) {
-        return result;
-    }
-    // Must be either reserved or function, not both; enforced by methods
-    if (isReservedSequence) {
-        // Methods enforce that the function name and options are unset
-        // if `setReserved()` is called, so if they were valid, that
-        // would indicate a bug.
-        U_ASSERT(!hasOptions && !hasFunctionName);
-        result = Operator(asReserved);
-    } else {
-        if (!hasFunctionName) {
-            // Neither function name nor reserved was set
-            // There is no default, so this case could occur if the
-            // caller creates a builder and doesn't make any calls
-            // before calling build().
-            errorCode = U_INVALID_STATE_ERROR;
-            return result;
-        }
-        result = Operator(functionName, options.build(errorCode));
-    }
-    return result;
+    return Operator(functionName, options.build(errorCode));
 }
 
-Operator::Operator(const Operator& other) noexcept : contents(other.contents) {}
+Operator::Operator(const Operator& other) noexcept
+    : name(other.name), options(other.options) {}
 
 Operator& Operator::operator=(Operator other) noexcept {
     swap(*this, other);
     return *this;
 }
 
 // Function call
-Operator::Operator(const FunctionName& f, const UVector& optsVector, UErrorCode& status) : contents(Callable(f, OptionMap(optsVector, status))) {}
+/*
+Operator::Operator(const FunctionName& f, const UVector& optsVector, UErrorCode& status)
+    : name(f), options(OptionMap(optsVector)) {}
+*/
```

Comment:
optional: remove?

Reply:
I'll remove this in my next PR so I can avoid the need for a reapproval :)

Reply:
Since you already lost the approval, can you maybe remove this now :-)
I am around, and promise to approve it instantly.
Thanks,
M.

Reply:
No problem if you don't feel like doing this on a Friday afternoon, just say so :-)

Reply:
Done!